### PR TITLE
Add Conversation API supporting. 

### LINKED ISSUE #28: Conversation API
As a Ruby developer I want to use the Conversations API through the Messagebird client

### PR DESCRIPTION
**Pull request by superstas**
Added following methods: Send; Start; Get (List & By ID)
It fixes #28

### CODE REVIEW

File: examples/conversation.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  
+  puts "  Channels                  :"
+  conversation.channels.each do |channel|
+  puts "    id                      : #{channel.id}"
+  puts "    name                    : #{channel.name}"
+  puts "    platformId              : #{channel.platformId}"
+  puts "    createdDatetime         : #{channel.createdDatetime}"
+  puts "    updatedDatetime         : #{channel.updatedDatetime}"
+  puts
+  end
```
Comment (khrt) [inline examples/conversation.rb]: Would it be better to have it this way:
```ruby
puts <<EOF
The following information was returned as a conversation object:"
  id                        : #{conversation.id}
...
EOF
```

File: lib/messagebird/conversation.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'messagebird/base'
+require 'messagebird/contact'
+require 'messagebird/conversation_channel'
+
+module MessageBird 
+  class Conversation < MessageBird::Base
+    attr_accessor :id, :contact, :channels, :status, :messages,
```
Comment (khrt) [inline lib/messagebird/conversation.rb]: I'm not really into Ruby anymore, but does it make sense to define `contact`, `channels` and others to which you've added setters as `attr_reader`? Maybe @marcelcorso or @epels  have an opinion on that?

File: lib/messagebird/conversation_client.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'net/https'
+require 'uri'
+require 'json'
+require 'messagebird/http_client'
+
+module MessageBird
+  class ConversationClient < HttpClient
+    def endpoint() 
+      CONVERSATIONS_ENDPOINT
```
Comment (khrt) [inline lib/messagebird/conversation_client.rb]: Let's define `CONVERSATIONS_ENDPOINT` on `MessageBird::Conversations` then.

File: lib/messagebird.rb [hunk 4]
```diff
@@ -2,8 +2,17 @@
...
+  ENDPOINT                = 'https://rest.messagebird.com/'
+  CONVERSATIONS_ENDPOINT  = 'https://conversations.messagebird.com/v1/'
+
+  CONVERSATION_STATUS_ACTIVE   = 'active'
+  CONVERSATION_STATUS_ARCHIVED = 'archived'
+
+  WEBHOOK_EVENT_CONVERSATION_CREATED = 'conversation.created'
+  WEBHOOK_EVENT_CONVERSATION_UPDATED = 'conversation.updated'
+  WEBHOOK_EVENT_MESSAGE_CREATED = 'message.created'
+  WEBHOOK_EVENT_MESSAGE_UPDATED = 'message.updated'
```
[the remark this answers is not in the document]
Comment (superstas) [reply lib/messagebird.rb]: I agree, moved

File: lib/messagebird/conversation_client.rb (hunk 3 above)
Comment (superstas) [reply lib/messagebird/conversation_client.rb]: Updated as well as ENDPOINT

**Pull request merged**